### ISSUE TITLE
Run inference by specifying a link of the trained model in config file

### ISSUE THREAD

**Issue by georgeyiasemis** [directgroup/direct#126]
Instead of downloading manually the weights of trained models locally, we want to specify the link of the trained model weights in the configuration file. When the program starts we want it to download the weights, load up the model, and start inference. The config ideally we'll look something like the below if the user wants to download automatically the weights:
```
...
checkpoint_url: https://<url-download-weights>/...
...
```
and like this otherwise
```
...
checkpoint_url: null
...
```

If a model checkpoint is specified  for inference (e.g. `--checkpoint 1000`)  with a given checkpoint_url (not `null`), we want this to throw a warning that this will be ignored.

Tasks for this feature:

- [x] Download weights from a given url. Throw an error if weights do not exist in the url (@jonasteuwen )
- [ ] Adjust config reading functions to work with this functionality (@georgeyiasemis )

**Comment by jonasteuwen** [comment]
I think we can just use "checkpoint_path" and allow a URL. The checkpointer can subsequently do a check if it is a real path or an URL and:
- if URL check if already downloaded (maybe use a `DIRECT_DOWNLOAD_DIR` as an environment variable)
- download if needed.

I think it should also work if you try to train a model and this variable is set! Then it should initialize from this checkpoint. So we should also find a way to ignore this setting. Thoughts?

**Comment by jonasteuwen** [comment]
See also #135. This will also make this easier.

**Comment by jonasteuwen** [comment]
@georgeyiasemis Perhaps you can now rebase from main, and use `model_checkpoint` in the config. The code infrastructure is now there to support URLs. There is also an `--initialization-checkpoint` flag so you can use that the code there to proceed with it all (or something similar). If that `model_checkpoint` is available, it means that the `--checkpoint` and things like that are not required.

**Comment by jonasteuwen** [comment]
Would be solved by #141

**Comment by jonasteuwen** [comment]
Let's close this for now, when you've implemented `direct predict` or something like that we can see how to do this for inference, as #141 implemented it for training.
